sources/node2/KalturaClientBase.js

```diff
@@ -334,16 +335,20 @@ class RequestBuilder extends kaltura.VolatileRequestData {
 					sessionId = response.headers[header];
 				}
 			}
-			client.debug('Response server [' + serverId + '] session [' + sessionId + ']: ' + data);
-
+			
 			let json;
-			try {
-				json = JSON.parse(data);
-			} catch (err) {
-				json = {
-					error: err
+			if (typeof data === 'string') {
+				client.debug('Response server (Str) [' + serverId + '] session [' + sessionId + ']: ' + data);
+				try {
+					json = JSON.parse(data);
+				} catch (err) {
+					json = data; //handle cases where Kaltura returns non JSON formatted responses like KS or serve

```

Comment:
We may want to output something to the log when the exception is caught.

Reply:
it's not an error, it's responses like ks that are simple string and not proper json.

Reply:
You'll note I didn't write `error`, I wrote `exception`. With the code as is now it is technically an exception, you use a `try{}catch{}` block for it..